src/components/SearchBar.js

```diff
@@ -278,14 +301,26 @@ class SearchBar extends React.Component<SearchBarDefaultProps, SearchBarProps, S
     }
   }
 
+  shareSearch() {
+    const username = AuthUtils.getUserName(AuthUtils.getSavedUser());
+    const signature = `\nfrom \n${username}`;
+    const message = this.props.shareMessage;
+    const searchLink = window.location.href;
```

Comment:
Can you get this data from [`withRouter`](https://reacttraining.com/react-router/web/api/history)?